src/synch.rs

```diff
@@ -0,0 +1,68 @@
+#![no_std]
+#![feature(core_intrinsics)]
+
+use core::sync::atomic::{AtomicBool, Ordering};
+use core::cell::UnsafeCell;
+use core::ops::{Deref, DerefMut};
+
+// A simple spinlock.
+pub struct SpinLock<T> {
+    lock: AtomicBool,
+    data: UnsafeCell<T>,
+}
+
+#[derive(Debug)]
+pub enum MutexError {
+    Poisoned,
+}
+
+// Safety: SpinLock can be safely sent across threads.
+unsafe impl<T> Sync for SpinLock<T> {}
+unsafe impl<T> Send for SpinLock<T> {}
+
+impl<T> SpinLock<T> {
+    // Creates a new spinlock.
+    pub const fn new(data: T) -> SpinLock<T> {
+        SpinLock {
+            lock: AtomicBool::new(false),
+            data: UnsafeCell::new(data),
+        }
+    }
+
+    // Acquires the spinlock, spinning until the lock is obtained.
+    pub fn lock(&self) -> SpinLockGuard<T> {
+        while self.lock.compare_exchange_weak(false, true, Ordering::Acquire, Ordering::Relaxed).is_err() {
```

Comment:
```suggestion
        while self
            .lock
            .compare_exchange_weak(false, true, Ordering::Acquire, Ordering::Relaxed)
            .is_err()
        {
```

Sorry, I haven't gotten around to adding a rustfmt check as part of a GitHub action, but if you make this change then the file should match what rustfmt expects.